### PR TITLE
Refactor make.jl

### PR DESCRIPTION
**Pull request by kellertuer**
I adopted what I already changed in Manopt.jl quite a while back
* more terminal ARGS can be passed now
* tutorials can be excluded from the docs when not rendering quarto – for simplicity when someone just wants to fix typos or work un tutorial-unrelated stuff.

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -2,6 +2,32 @@
 #
 #
 
+if "--help" ∈ ARGS
+    println(
+        """
+docs/make.jl
+
+Render the `Manopt.jl` documentation with optional arguments
```
Comment (mateuszbaran) [inline docs/make.jl:10]:
```suggestion
Render the `Manifolds.jl` documentation with optional arguments
```

File: docs/make.jl [hunk 2]
```diff
@@ -2,6 +2,32 @@
...
+* `--help`              - print this help and exit without rendering the documentation
+* `--prettyurls`        – toggle the prettyurls part to true (which is otherwise only true on CI)
+* `--quarto`            – run the Quarto notebooks from the `tutorials/` folder before generating the documentation
+  this has to be run locally at least once for the `tutorials/*.md` files to exist that are included in
+  the documentation (see `--exclude-tutorials`) for the alternative.
+  If they are generated once they are cached accordingly.
+  Then you can spare time in the rendering by not passing this argument.
+  If quarto is not run, some tutorials are generated as empty files, since they
+  are referenced from within the documentation. These are currently
+  `Optimize.md` and `ImplementOwnManifold.md`.
```
Comment (mateuszbaran) [inline docs/make.jl:25]: Maybe let's delete the last sentence here because it's likely to be outdated soon.

**Comment by kellertuer** [comment]
And our News.md is out of sync, it states news for 0.10.6 which it seems are maybe part of 10.5? Will check tomorrow.

This does not need to be released as its own version and can chill on main for a while though. It is just nicer for new contributors.

**Comment by mateuszbaran** [comment]
I forgot to bump version in Project.toml to 0.10.6 in the last PR so maybe let's release 0.10.6 after merging this PR?

**Comment by kellertuer** [comment]
Sure, I adapted news and project.toml.

**Review by mateuszbaran — APPROVED** [review]

**Comment by mateuszbaran** [comment]
I have no idea, maybe someone on Slack or discourse could help?

**Comment by kellertuer** [comment]
Asked on slack never got an answer.

**Comment by mateuszbaran** [comment]
:disappointed:

**Pull request merged**